modules/seqexec/server/src/main/scala/seqexec/server/InstrumentActions.scala

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2016-2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.server
+
+import cats.MonadError
+import cats.data.NonEmptyList
+import cats.implicits._
+import fs2.Stream
+import io.chrisdavenport.log4cats.Logger
+import seqexec.model.ActionType
+import seqexec.model.dhs.ImageFileId
+import seqexec.engine.Action
+import seqexec.engine.Action.ActionState
+import seqexec.engine.ParallelActions
+import seqexec.engine.Result
+
+/**
+  * Algebra to generate actions for an observation.
+  * Most instruments behave the same but in some cases we need to customize
+  * behavior. The two prime examples are:
+  * GPI A&C
+  * GMOS N&S
+  * In both cases the InstrumentActions for the instrument can provide the correct behavior
+  */
+trait InstrumentActions[F[_]] {
```

Comment:
This is a key change. We are now delegating the construction of actions to each instrument